full-moon/src/ast/visitors.rs

```diff
@@ -431,3 +431,137 @@ impl<'a> VisitMut<'a> for LocalAssignment<'a> {
         self
     }
 }
+
+impl<'a> Visit<'a> for GenericFor<'a> {
+    fn visit<V: Visitor<'a>>(&self, visitor: &mut V) {
+        visitor.visit_generic_for(self);
+        self.for_token.visit(visitor);
+
+        let mut type_specifiers;
+
+        #[cfg(feature = "roblox")]
+        {
+            type_specifiers = self.type_specifiers();
+        }
+
+        #[cfg(not(feature = "roblox"))]
+        {
+            // TODO: Option<!>, and implement Visit for !
+            type_specifiers = std::iter::repeat::<Option<Self>>(None);
+        }
+
+        for name in &self.names {
+            name.visit(visitor);
+            type_specifiers.next().visit(visitor);
+        }
+
+        self.in_token.visit(visitor);
+        self.expr_list.visit(visitor);
+        self.do_token.visit(visitor);
+        self.block.visit(visitor);
+        self.end_token.visit(visitor);
+
+        visitor.visit_generic_for_end(self);
+    }
+}
+
+impl<'a> VisitMut<'a> for GenericFor<'a> {
+    fn visit_mut<V: VisitorMut<'a>>(mut self, visitor: &mut V) -> Self {
+        self = visitor.visit_generic_for(self);
+        self.for_token = self.for_token.visit_mut(visitor);
+
+        let mut type_specifiers;
```

Comment:
```suggestion
        let type_specifiers;
```

Reply:
Same as above, the iterator is being mutated